mbuild/formats/gomc_conf_writer.py

```diff
@@ -2013,43 +2007,57 @@ def __init__(
             charmm_object.all_res_unique_atom_name_dict
         )
 
-        self.x_dim_box_0 = (
-            charmm_object.box_0.Lx * 10
-        )  # times 10 to convert from nm to Angstroms
-        self.y_dim_box_0 = (
-            charmm_object.box_0.Ly * 10
-        )  # times 10 to convert from nm to Angstroms
-        self.z_dim_box_0 = (
-            charmm_object.box_0.Lz * 10
-        )  # times 10 to convert from nm to Angstroms
-
-        print(
-            "charmm_object.filename_box_1 = ",
-            format(str(charmm_object.filename_box_1)),
-        )
-        print(
-            "type(charmm_object.filename_box_1) = ",
-            format(str(type(charmm_object.filename_box_1))),
-        )
+        # Get and test and make sure vectors are not too large for the 16 spaces allotted
+        box_vectors_char_limit = 16
+        self.box_0_vectors = charmm_object.box_0_vectors
+        box_0_vectors_char_ok = True
+        for x_i in range(0, 3):
+            for y_i in range(0, 3):
+                if (
+                    len(str(self.box_0_vectors[x_i][y_i]))
+                    > box_vectors_char_limit
+                ):
+                    box_0_vectors_char_ok = False
 
+        if box_0_vectors_char_ok == False:
+            self.input_error = True
+            print_error_message = (
+                "ERROR: At lease one of the individual box {} vectors are too large "
+                "or greater than {} characters."
+                "".format(
+                    0,
+                    box_vectors_char_limit,
+                )
+            )
+            raise ValueError(print_error_message)
         if self.ensemble_type in ["GEMC_NVT", "GEMC_NPT", "GCMC"]:
             if (
                 charmm_object.filename_box_1 is not None
                 and isinstance(charmm_object.filename_box_1, str) is True
             ):
-                self.x_dim_box_1 = (
-                    charmm_object.box_1.Lx * 10
-                )  # times 10 to convert from nm to Angstroms
-                self.y_dim_box_1 = (
-                    charmm_object.box_1.Ly * 10
-                )  # times 10 to convert from nm to Angstroms
-                self.z_dim_box_1 = (
-                    charmm_object.box_1.Lz * 10
-                )  # times 10 to convert from nm to Angstroms
+                self.box_1_vectors = charmm_object.box_1_vectors
+                box_1_vectors_char_ok = True
```

Comment:
Since you use this at least twice, it makes sense to break this into a method i think

```python3
def _check_box_vectors_char_limit(vectors, char_limit):
    for x_i in range(0, 3):
        for y_i in range(0, 3):
            if (
                len(str(vectors[x_i][y_i]))
                > char_limit
            ):
                return False
    return True
```

Reply:
Added a new function as you listed to the gomc_conf_writer.py and applied it in the 2 locations where it was being used.